src/DotLiquid/StandardFilters.cs

```diff
@@ -95,6 +95,18 @@ public static string UrlEncode(string input)
                 ? input
                 : Uri.EscapeDataString(input);
         }
+		
+		/// <summary>
+        /// convert a input string to URLDECODE
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+		public static string UrlDecode(string input)
+        {
+            return input == null
+                ? input
+                : Uri.UnescapeDataString(input);

```

Comment:
[`UnescapeDataString`](https://docs.microsoft.com/en-us/dotnet/api/system.uri.unescapedatastring?view=netframework-4.7.2) doesn't handle the `+` to space conversion and I think that Liquid does since the [`url_encode` filter](http://shopify.github.io/liquid/filters/url_encode/) converts the space to `+`. If it is true, a `.Replace('+',' ')` should be added.

Reply:
Yes, that's true. I replace with HttpUtility.UrlEncode / UrlDecode